Add ability to query advertiserStories by externalId

B77Mills opened the pull request:
Requires https://github.com/parameter1/fortnight-graph/pull/20
@todo ensure externalId || advertiserId is sent. 

![watt-company-brand-insights](https://github.com/parameter1/fortnight-graph/assets/3845869/a50a73a4-2009-4f2e-8667-cfe36521494e)

B77Mills committed:
Add ability to query advertiserStories by exteriorId

@todo ensure externalId || advertiserId is sent.

zarathustra323 commented on src/graph/resolvers/story.js:
```diff
@@ -216,14 +216,20 @@ module.exports = {
         excludeStoryIds,
       } = input;
+      // set id(advertiserId) based externalId look up or by it bein passed.
+      const id = (externalId && !advertiserId)
```
This should throw a user input error in two cases:

If both the advertiser and the external IDs are provided 

If neither of the IDs are provided. 

There are some examples of this in some of the base CMS graphql resolvers.

B77Mills committed:
Throw an error when no advertiser is found.

Throw a No advertiser found for ID ‘${id}’ error when a advertiser is not found

zarathustra323 commented on src/graph/resolvers/story.js:
```diff
@@ -216,14 +216,24 @@ module.exports = {
+        : advertiserId;
+
+      if (!id) {
+        throw new Error(`No advertiser found for ID '${id}'`);
```
What I meant was to throw two individual user input errors for each case (both IDs being provided _or_ neither ID being provided). See: https://github.com/parameter1/base-cms/blob/44fd13f31c67deae3513f2724a9a277d49869200/services/graphql-server/src/graphql/resolvers/platform/content.js#L653

B77Mills committed:
Add conditional error handleing

Add conditional error handelin for when both, neither or invalid externalId/advertiserId are. provided

zarathustra323 commented on src/graph/resolvers/story.js:
```diff
@@ -216,14 +216,30 @@ module.exports = {
         excludeStoryIds,
       } = input;
+
+      if (!advertiserId && !externalId) throw new Error('Either a advertiserId or externalId input must be provided.');
```
These should throw `UserInputError` instances -- see previous comment on how to `require` the error :)

zarathustra323 commented on src/graph/resolvers/story.js:
```diff
@@ -216,14 +216,30 @@ module.exports = {
+      if (advertiserId && externalId) throw new Error('You cannot provide both advertiserId and externalId as input.');
+
+      // set id(advertiserId) based externalId look up or by it bein passed.
+      const id = (externalId && !advertiserId)
```
Clean this up now that you're checking the inputs...

```js
let id = advertiserId;
if (externalId) { // load the advertiser by external id and set id to advertiser.id }
```

zarathustra323 commented on src/graph/resolvers/story.js:
```diff
@@ -216,14 +216,30 @@ module.exports = {
+        ? await Advertiser.find({ externalId })
+        : advertiserId;
+
+      if (!id && externalId) {
```
Change this throw... `if (!id) {  throw new Error(`No advertiser found with user external ID '${externalId}'`); }`

B77Mills committed:
Cleanup prop validation error handling

B77Mills committed:
update maessage

B77Mills committed:
More cleanup to error handelingg
